README.md

```diff
@@ -1,54 +1,52 @@
 
-For developers looking to build services locally (e.g., the coordinator), see our detailed [Local Development Guide](docs/local-development-guide.md).
+**火 Karma System**: A reputation-based mechanism where users earn Karma soulbound tokens through positive network participation. Users will have different levels of daily gasless transaction quota depending on their Karma amount. Contract code implementation can be found [here](https://github.com/vacp2p/staking-reward-streamer).
 
-## Looking for the Linea code?
+### Key Features
 
-Linea's stack is made up of multiple repositories, these include:
+- **🆓 Zero Gas Transactions**: Users with sufficient Karma balance can submit transactions without paying gas
+- **🛡️ Rate Limiting**: RLN-based rate limiting prevents spam and abuse through cryptographic proofs
+- **💰 Premium Gas Bypass**: Users who exceed their quota can bypass restrictions with premium gas payments
+- **📊 Reputation-Based Access**: Karma tokens determine user privileges and transaction quotas
 
-- [linea-monorepo](https://github.com/Consensys/linea-monorepo): The main repository for the Linea stack & network
-- [linea-besu](https://github.com/Consensys/linea-besu): Fork of Besu to implement the Linea-Besu client
-- [linea-sequencer](https://github.com/Consensys/linea-sequencer): A set of Linea-Besu plugins for the sequencer and RPC nodes
-- [linea-tracer](https://github.com/Consensys/linea-tracer): Linea-Besu plugin which produces the traces that the constraint system applies and that serve as inputs to the prover
-- [linea-constraints](https://github.com/Consensys/linea-constraints): Implementation of the constraint system from the specification
-- [linea-specification](https://github.com/Consensys/linea-specification): Specification of the constraint system defining Linea's zkEVM
+### Architecture Components
 
-Linea abstracts away the complexity of this technical architecture to allow developers to:
+#### Prover Components
+- [**TxForwarder Transaction Pool Validator**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txpoolvalidation/validators/RlnProverForwarderValidator.java): Forwards incoming transaction data to the RLN prover service to generate RLN proofs
+- [**Enhanced LineaEstimateGas RPC**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/rpc/methods/LineaEstimateGas.java): Dynamically provides zero gas or premium gas estimates based on real-time user karma and usage quotas
+- [**Karma Service Integration**](besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txpoolvalidation/shared/KarmaServiceClient.java): Real-time user tier and quota checking via gRPC
 
-- [Bridge tokens](https://docs.linea.build/developers/guides/bridge)
-- [Deploy a contract](https://docs.linea.build/developers/quickstart/deploy-smart-contract)
-- [Run a node](https://docs.linea.build/developers/guides/run-a-node)
+#### Verifier (Sequencer) Components
```

Comment:
Sequencer components sounds better than verifier, since there are other components besides verification

Reply:
I think this section is missing the rust code for RLN JNI bridge

Reply:
[b7f9893](https://github.com/status-im/status-network-monorepo/pull/4/commits/b7f9893f981ce6d871073dda682c5308dacbf019)
- removed `verifier`
- added RLN JNI bridge